Home page header markup and corresponding styles

IanaKarpenko opened the pull request.

IanaKarpenko committed:
Main page header+styles are done

Completed header for home page:
- Defined home page header markup;
- Defined home page heade stylesr;
- Defined home page logo component;
- Defined home page styles variables.

IanaKarpenko committed:
Corrected background images

Corrected home page header styles to display correct background images

IanaKarpenko committed:
Completed styles for home page header

Corrected styles for home page header

IanaKarpenko committed:
Corrected styles/markup

Corrected styles/markup according to BEM

AlexCanDoIt reviewed (commented):
Оптимизировать все изображения

AlexCanDoIt commented on src/index.html:
```diff
@@ -10,10 +10,14 @@
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500&display=swap" rel="stylesheet">
   </head>
   <body>
-    <!-- Так добавляются фрагменты в главные HTML-файлы страниц. -->
-    <include src="./partials/example.html"></include>
+    <!-- Так добавляются фрагменты в главные HTML-файлы страниц. 
+    <include src="./partials/example.html"></include>-->
```
Примеры можно убрать вообще

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+<section class="header">
+    <div class="header__container">
+        <nav class="header__nav">
+            <a class="logo link">
+                <svg width="24" height="24" class="logo__svg">
+                    <use href="./images/sprite.svg#film"></use>
+                </svg>
+                <span class="logo__text">Filmoteka</span>
+            </a>
```
Вот тут добавить класс "header__logo" (class="logo link header__logo") и стили позиционирования лого в хедере по этому новому классу писать
У тебя должен получиться отдельный паршал в сасс на лого без маржинов падингов и т.п. А в стилях на паршале сасс хедера - позиционирование логотипа

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+                </svg>
+                <span class="logo__text">Filmoteka</span>
+            </a>
+            <ul class="header__navlist list">
```
Исправить "header__navlist list" на "header__nav-list list". И дальше в коде тоже

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+                <span class="logo__text">Filmoteka</span>
+            </a>
+            <ul class="header__navlist list">
+                <li class="header__navlist-elem home">
```
Исправить класс на "header__item home" и дальше в коде по аналогии

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+            </a>
+            <ul class="header__navlist list">
+                <li class="header__navlist-elem home">
+                    <a class="header__navlist-elemlink">Home</a>
```
Исправить класс на "header__link" и дальше в коде по аналогии

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+            </ul>
+        </nav>
+
+        <div class="header__search-wrapper">
```
Исправить класс на "header__form-container"

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+        </nav>
+
+        <div class="header__search-wrapper">
+            <form class="header__search-form" id="header__search-form">
```
Исправить класс на "header__form form". Форму поиска можно сделать отдельным компонентом (как лого) и потом переиспользовать. Второй вариант, к классам что я написал ниже перед словом form добавь header (header-form__), если считаешь, что этот инпут (форма) больше никогда не пригодиться

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+
+        <div class="header__search-wrapper">
+            <form class="header__search-form" id="header__search-form">
+                <input type="text" class="header__search-input" name="header__search-input" autocomplete="off"
```
Исправить класс на "form__input"

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+            <form class="header__search-form" id="header__search-form">
+                <input type="text" class="header__search-input" name="header__search-input" autocomplete="off"
+                    placeholder="Поиск фильмов" />
+                <span class="header__search-spansizer">
```
Исправить класс на "header__form-text"

AlexCanDoIt replied:
Исправить класс на "form__svg-container"

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+            <form class="header__search-form" id="header__search-form">
+                <input type="text" class="header__search-input" name="header__search-input" autocomplete="off"
+                    placeholder="Поиск фильмов" />
+                <span class="header__search-spansizer">
+                    <svg width="24" height="24" class="header__search-svg">
+                        <use href="./images/sprite.svg#search"></use>
+                    </svg>
+                </span>
```
Исправить класс на "form__svg-container"

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+                <input type="text" class="header__search-input" name="header__search-input" autocomplete="off"
+                    placeholder="Поиск фильмов" />
+                <span class="header__search-spansizer">
+                    <svg width="24" height="24" class="header__search-svg">
```
Исправить класс на "form__svg"

AlexCanDoIt commented on src/partials/_home-header.html:
```diff
@@ -0,0 +1,35 @@
+                        <use href="./images/sprite.svg#search"></use>
+                    </svg>
+                </span>
+                <p class="header__errmsg header__errmsg-hide">
```
Исправить класс на "form__text". А для хиддена у нас есть утилита, вернее будет. Добавь её заодно в сасс/утилиты/визуали-хидден и туда положи этот селектор (и его же и используй что бы стрятать сообщение):

.visually-hidden {
  position: absolute !important;
  clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
  clip: rect(1px, 1px, 1px, 1px);
  padding: 0 !important;
  border: 0 !important;
  height: 1px !important;
  width: 1px !important;
  overflow: hidden;
}

IanaKarpenko committed:
Optimized images

Optimized images

IanaKarpenko committed:
Completed corrections after comments

1. Deleted example partial from index.html
2. Added header__logo
3. Corrected "header__navlist list" to "header__nav-list list"
4. Corrected class name to "header__item home"
5. Corrected class to "header__link"
6. Corrected class to "header__form-container"
7. Corrected class name to  "form__svg"
8. Corrected class name to "form__text".
9. Used visually-hidden.scss to hide errmrg

AlexCanDoIt reviewed (approved):
Принято

AlexCanDoIt reviewed (approved):
Approve
